pkg/mco/mcpList.go

```diff
@@ -8,83 +8,73 @@ import (
 	"github.com/golang/glog"
 	"github.com/openshift-kni/eco-goinfra/pkg/clients"
 
-	mcov1 "github.com/openshift/machine-config-operator/pkg/apis/machineconfiguration.openshift.io/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 	"k8s.io/apimachinery/pkg/util/wait"
 )
 
-// MCPListBuilder provides struct for MachineConfigPoolList object which contains connection to cluster
-// and MachineConfigPoolList definitions.
-type MCPListBuilder struct {
-	// MachineConfigPoolList definition. Used to create MachineConfigPoolList object with minimum
-	// set of required elements.
-	ObjectList *mcov1.MachineConfigPoolList
-	apiClient  *clients.Settings
-	// machineConfigSelector label.
-	mcSelector string
-}
-
-// NewMCPListBuilder method creates new instance of MCPListBuilder.
-func NewMCPListBuilder(apiClient *clients.Settings, mcpLabelSelector ...map[string]string) *MCPListBuilder {
-	glog.V(100).Infof(
-		"Initializing new MCPListBuilder structure with the following params: %v", mcpLabelSelector)
-
-	builder := &MCPListBuilder{
-		apiClient: apiClient,
-	}
+func ListMCP(apiClient *clients.Settings, mcpLabelSelector ...map[string]string) ([]*MCPBuilder, error) {
```

Comment:
I think you should just have options v1.ListOptions as an argument rather than mcpLabelSelector. This is what we currently do in other pkgs